source/partial-deep.d.ts

```diff
@@ -30,6 +30,8 @@ settings = applySavedSettings({textEditor: {fontWeight: 500}});
 */
 export type PartialDeep<T> = T extends Primitive
 	? Partial<T>
+	: T extends Date
+	? T
```

Comment:
Rather than setting `Date` here explicitly, could you maybe do something like:

```suggestion
export type PartialDeep<T, ExceptTypes = never> = T extends Primitive
	? Partial<T>
	: T extends ExceptTypes
	? T
```

Then someone can chose to exclude eg. `Date` when that's a good idea, or eg a collection like `PartialDeep<Foo, MyBuiltInTypes>`